### ISSUE TITLE
PPO + gSDE / A2C + gSDE / SAC + gSDE

### ISSUE THREAD

**Issue by araffin** [koulakis/reacher-deep-reinforcement-learning#3]
Hello,
nice project.

I was wondering if you had considered trying generalized state-dependent exploration (gSDE) and the tuned hyperparameters from the zoo:

A2C: https://github.com/DLR-RM/rl-baselines3-zoo/blob/master/hyperparams/a2c.yml#L126
PPO: https://github.com/DLR-RM/rl-baselines3-zoo/blob/master/hyperparams/ppo.yml#L137
SAC: https://github.com/DLR-RM/rl-baselines3-zoo/blob/master/hyperparams/sac.yml#L141

In a nutshell, gSDE is a different exploration strategy that is made for the continuous action case. It allows to train RL agent directly on real robot for instance.
Cf paper: https://arxiv.org/abs/2005.05719

If you do so, please upgrade your SB3 version, as we pushed a fix recently for PPO+gSDE 
and keep me up to date, I'm interested by the results ;)
See https://github.com/DLR-RM/stable-baselines3/releases for the last release

**Comment by koulakis** [comment]
Hello Antonin,

Thanks for the hints. Also kudos for stable baselines 3; it has been a real pleasure using this framework up to now!

No, I haven't tried either the zoo parameters or gSDE yet, but I am curious to try both. The agents reach an almost optimal solution in terms of reward, but would be great to achieve faster convergence and also reduce the shaking behavior of the joints.

My plan is to: 
- re-run each algorithm (A2C, PPO, TD3 and maybe try out SAC) with the half cheetah hyper-parameters you proposed and fine-tune them a bit for reacher
- repeat the above experiments with the addition of gSDE and check its effect on convergence and the final agents' shaking behavior

I will keep you posted when I make progress. If I am lucky I might get some time to work on this one this weekend :)

Cheers,
Marios

**Comment by araffin** [comment]
>achieve faster convergence 

using less samples? i would definitely try SAC with tuned hyperparameters for that ;)
